tests/main/interfaces-network-status/task.yaml

```diff
@@ -0,0 +1,67 @@
+summary: Ensure that the network-status interface works.
+
+details: |
+    The network-status interface allows a snap to monitor network connection status.
+
+    A snap which defines the network-status plug must be shown in the interfaces list.
+    The plug is autoconnected on install and, as usual, must be able to be disconnect/reconnect.
+
+    The snap is also declaring a plug on this interface must be able to ask for its status.
+
+# dbus-launch not supported in ubuntu-core
+systems: [-ubuntu-core-16-*]
+
+prepare: |
+    . "$TESTSLIB/dbus.sh"
+    . "$TESTSLIB/dirs.sh"
+
+    echo "Given a snap declaring a plug on the network-status interface is installed"
+    snap install test-snapd-network-status-provider
+
+    echo "And the provider dbus loop is started"
+    start_dbus_unit $SNAP_MOUNT_DIR/bin/test-snapd-network-status-provider.provider
+
+restore: |
+    rm -f getstate.error
+
+    . "$TESTSLIB/dbus.sh"
+    stop_dbus_unit
+
+execute: |
+    CONNECTED_PATTERN="^test-snapd-network-status-provider:network-status-test +test-snapd-network-status-provider:network-status"
+    DISCONNECTED_PATTERN="^- +test-snapd-network-status-provider:network-status$"
+
+    echo "The interface is connected by default"
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then wait until the dbus name is properly reserved"
+    for i in $(seq 10); do
+        if ! test-snapd-network-status-provider.consumer GetVersion | MATCH "my-ap-version"; then
+            sleep 1
+        else
+            break
+        fi
+    done
+
+    echo "Check the network-status is working from the consumer app in the snap"
+    test-snapd-network-status-provider.consumer GetState | MATCH "my-ap-state"
+    test-snapd-network-status-provider.consumer GetVersion | MATCH "my-ap-version"
```

Comment:
Ah, ok, you can ignore the above comment as we will fail here eventually if it didn't match in the loop. That's ok.